src/design/introspection.stanza

```diff
@@ -89,3 +90,36 @@ public defn get-element-ports (obj:JITXObject) -> [JITXObject, JITXObject] :
     (_:None):
       [obj.p[1], obj.p[2]]
 
+doc: \<DOC>
+Get the net associated with a module or component port
+
+@param pt Port of a component or module.
+@return If a connection is found - this function returns the Net as
+a JITXObject. If no connection is found - then None().
+<DOC>
+public defn get-connected-net (pt:JITXObject) -> Maybe<JITXObject>:
+  inside pcb-module:
+    for n in nets(self) first :
+      if connected?([pt, n]):
+        One(n)
+      else:
+        None()
```

Comment:
Philosophically, shouldn't these introspection functions (in this file) be part of jitx-client? I have no issue with them in JSL, but it seems to me that we are going down the path like ocdb where very useful functions ended up in user code versus the client.

Reply:
Ya - thats fair. 

I think prior to the query API - the port/pad definitions for `p[1]/p[2]` and `c/a` were in OCDB and so it didn't make sense to put them in JITX proper. 

I think with the refactor @plasmajitx  has done - it may now make more sense to put some of those types of functions in JITX. 

There are others here that probably should be in JITX. 

I think we need to talk about organization of that content - I can create a ticket.